### ISSUE TITLE
Operator pruning unknown resources extremely slow

### ISSUE THREAD

**Issue by howardjohn** [istio/istio#19816]
```
2019-12-27T17:01:06.096070Z     info    end pruning
2019-12-27T17:01:06.096074Z     info    end deleting resources
2019-12-27T17:01:06.112070Z     info    Reconciling IstioControlPlane
2019-12-27T17:01:06.112199Z     info    Deleting IstioControlPlane
2019-12-27T17:01:06.112216Z     info    begin deleting resources
2019-12-27T17:01:06.112221Z     info    begin pruning
2019-12-27T17:01:06.988056Z     warn    retrieving resources to prune type authentication.istio.io/v1alpha1, Kind=Policy: the server could not find the requested resource not found
2019-12-27T17:01:07.289312Z     warn    retrieving resources to prune type certmanager.k8s.io/v1beta1, Kind=Certificate: no matches for kind "Certificate" in version "certmanager.k8s.io/v1beta1" not found
2019-12-27T17:01:09.089291Z     warn    retrieving resources to prune type certmanager.k8s.io/v1beta1, Kind=Challenge: no matches for kind "Challenge" in version "certmanager.k8s.io/v1beta1" not found
2019-12-27T17:01:10.889458Z     warn    retrieving resources to prune type certmanager.k8s.io/v1beta1, Kind=Issuer: no matches for kind "Issuer" in version "certmanager.k8s.io/v1beta1" not found
2019-12-27T17:01:12.692409Z     warn    retrieving resources to prune type certmanager.k8s.io/v1beta1, Kind=Order: no matches for kind "Order" in version "certmanager.k8s.io/v1beta1" not found
2019-12-27T17:01:14.512474Z     warn    retrieving resources to prune type config.istio.io/v1alpha2, Kind=adapter: no matches for kind "adapter" in version "config.istio.io/v1alpha2" not found
2019-12-27T17:01:16.289173Z     warn    retrieving resources to prune type config.istio.io/v1alpha2, Kind=attributemanifest: no matches for kind "attributemanifest" in version "config.istio.io/v1alpha2" not found
2019-12-27T17:01:18.089655Z     warn    retrieving resources to prune type config.istio.io/v1alpha2, Kind=handler: no matches for kind "handler" in version "config.istio.io/v1alpha2" not found
2019-12-27T17:01:19.892489Z     warn    retrieving resources to prune type config.istio.io/v1alpha2, Kind=instance: no matches for kind "instance" in version "config.istio.io/v1alpha2" not found
2019-12-27T17:01:21.692170Z     warn    retrieving resources to prune type config.istio.io/v1alpha2, Kind=HTTPAPISpec: no matches for kind "HTTPAPISpec" in version "config.istio.io/v1alpha2" not found
2019-12-27T17:01:23.493298Z     warn    retrieving resources to prune type config.istio.io/v1alpha2, Kind=HTTPAPISpecBinding: no matches for kind "HTTPAPISpecBinding" in version "config.istio.io/v1alpha2" not found
2019-12-27T17:01:25.293822Z     warn    retrieving resources to prune type config.istio.io/v1alpha2, Kind=QuotaSpec: no matches for kind "QuotaSpec" in version "config.istio.io/v1alpha2" not found
2019-12-27T17:01:27.089588Z     warn    retrieving resources to prune type config.istio.io/v1alpha2, Kind=QuotaSpecBinding: no matches for kind "QuotaSpecBinding" in version "config.istio.io/v1alpha2" not found
2019-12-27T17:01:28.892578Z     warn    retrieving resources to prune type config.istio.io/v1alpha2, Kind=rule: no matches for kind "rule" in version "config.istio.io/v1alpha2" not found
2019-12-27T17:01:30.689837Z     warn    retrieving resources to prune type config.istio.io/v1alpha2, Kind=template: no matches for kind "template" in version "config.istio.io/v1alpha2" not found
2019-12-27T17:01:32.492908Z     warn    retrieving resources to prune type networking.istio.io/v1alpha3, Kind=DestinationRule: no matches for kind "DestinationRule" in version "networking.istio.io/v1alpha3" not found
2019-12-27T17:01:34.292485Z     warn    retrieving resources to prune type networking.istio.io/v1alpha3, Kind=EnvoyFilter: no matches for kind "EnvoyFilter" in version "networking.istio.io/v1alpha3" not found
2019-12-27T17:01:36.090013Z     warn    retrieving resources to prune type networking.istio.io/v1alpha3, Kind=Gateway: no matches for kind "Gateway" in version "networking.istio.io/v1alpha3" not found
2019-12-27T17:01:37.891573Z     warn    retrieving resources to prune type networking.istio.io/v1alpha3, Kind=ServiceEntry: no matches for kind "ServiceEntry" in version "networking.istio.io/v1alpha3" not found
2019-12-27T17:01:39.690067Z     warn    retrieving resources to prune type networking.istio.io/v1alpha3, Kind=Sidecar: no matches for kind "Sidecar" in version "networking.istio.io/v1alpha3" not found
2019-12-27T17:01:41.492317Z     warn    retrieving resources to prune type networking.istio.io/v1alpha3, Kind=VirtualService: no matches for kind "VirtualService" in version "networking.istio.io/v1alpha3" not found
2019-12-27T17:01:43.292464Z     warn    retrieving resources to prune type rbac.istio.io/v1alpha1, Kind=ClusterRbacConfig: no matches for kind "ClusterRbacConfig" in version "rbac.istio.io/v1alpha1" not found
2019-12-27T17:01:45.092543Z     warn    retrieving resources to prune type rbac.istio.io/v1alpha1, Kind=RbacConfig: no matches for kind "RbacConfig" in version "rbac.istio.io/v1alpha1" not found
2019-12-27T17:01:46.891665Z     warn    retrieving resources to prune type rbac.istio.io/v1alpha1, Kind=ServiceRole: no matches for kind "ServiceRole" in version "rbac.istio.io/v1alpha1" not found
2019-12-27T17:01:48.692186Z     warn    retrieving resources to prune type rbac.istio.io/v1alpha1, Kind=ServiceRoleBinding: no matches for kind "ServiceRoleBinding" in version "rbac.istio.io/v1alpha1" not found
2019-12-27T17:01:50.492833Z     warn    retrieving resources to prune type security.istio.io/v1beta1, Kind=AuthorizationPolicy: no matches for kind "AuthorizationPolicy" in version "security.istio.io/v1beta1" not found
2019-12-27T17:01:52.292656Z     warn    retrieving resources to prune type security.istio.io/v1beta1, Kind=RequestAuthentication: no matches for kind "RequestAuthentication" in version "security.istio.io/v1beta1" not found
2019-12-27T17:01:54.089729Z     warn    retrieving resources to prune type certmanager.k8s.io/v1beta1, Kind=ClusterIssuer: no matches for kind "ClusterIssuer" in version "certmanager.k8s.io/v1beta1" not found
2019-12-27T17:01:55.889936Z     warn    retrieving resources to prune type authentication.istio.io/v1alpha1, Kind=MeshPolicy: no matches for kind "MeshPolicy" in version "authentication.istio.io/v1alpha1" not found
```

Look at the timing -- its almost 2s per resource. I think its slower because the things its pruning do not exist (releated: https://github.com/istio/istio/issues/19815)

**Comment by ostromart** [comment]
@howardjohn pointed out that go client API requests are rate controlled to a low value by default. We should prioritize trying out a faster rate since this will also speed up the control loop greatly.

**Comment by kong62** [comment]
I  have  same problem in  istio  1.5.0

**Comment by ostromart** [comment]
See my unsuccessful attempt to bump this up in https://github.com/istio/istio/pull/21288.
If anyone's interested in delving deeper into this and doing some experimentation, please feel free to take over that PR.

**Comment by richardwxn** [comment]
@ostromart let me take this over then

**Comment by howardjohn** [comment]
Fixed in #23314
